beam/docs/zebra_printing.md

```diff
@@ -0,0 +1,178 @@
+# Zebra Printing
+
+To create a Zebra print format, you need the following documents:
+- A ZPL Print Format made against Handling Units that uses the available Jinja utility functions to generate ZPL code.
+- A document Print Format that uses the free Labelary API to convert the above ZPL code and generate a preview of the print output for the linked document.
+
+### ZPL Code Generation
+
+Currently, only three types of printable ZPL data can be generated with utilties within Beam:
+- `Text`
+- `Barcode`
+- `Label`
+
+Beam uses the [py-zebra-zpl](https://github.com/mtking2/py-zebra-zpl) library to generate the above types, as it provides a basic interface to create ZPL code using Python objects. Please refer to the library's documentation for more information on how to use it.
+
+In addition, Beam exposes the following Jinja functions to be used within a Print Format:
+
+---
+
+#### `barcode128`
+
+Generate a [Code 128](https://en.wikipedia.org/wiki/Code_128) barcode image. It takes the following arguments:
+
+- `barcode_text`: The text to be encoded in the barcode. Required.
+
+##### Example
+```jinja
+{{ barcode128('123456') }}
+```
+
+---
+
+#### `formatted_zpl_barcode`
+
+Generate a formatted ZPL barcode. It takes the following arguments:
+
+- `barcode_text`: The text to be encoded in the barcode. Required.
+
+##### Example
+```jinja
+{{ formatted_zpl_barcode('123456') }}
+```
+
+---
+
+#### `formatted_zpl_label`
+
+Generate a formatted ZPL label object. It takes the following arguments:
+
+- `width`: The width of the label in dots. Required.
+- `height`: The height of the label in dots. Required.
+- `dpi`: The DPI of the printer. Defaults to 203.
+  - Visit the [official documentation](https://supportcommunity.zebra.com/s/article/000026166) to determine the DPI for your Zebra printer model.
+- `print_speed`: The print speed of the printer in inches per second (ips). Defaults to 2.
+  - Slower print speeds typically yield better print quality.
+  - Visit the [official documentation](https://supportcommunity.zebra.com/s/article/Setting-the-Print-Speed-via-ZPL) to determine the acceptable print speed values for your Zebra printer model.
+- `copies`: The number of copies to print. Defaults to 1.
+
+##### Example
+```jinja
+{% set label = formatted_zpl_label(width=6*203, height=4*203, dpi=203) %}
```

Comment:
Is it worth mentioning that the math here is inches * DPI?

Reply:
I think so. I've added it to the docs for this function directly.